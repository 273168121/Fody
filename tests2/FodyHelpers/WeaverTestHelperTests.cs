using System.Collections.Generic;
using System.IO;
using Fody;
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
using ObjectApproval;
#endif
using Xunit;

//using Xunit;

#pragma warning disable 618
public class WeaverTestHelperTests : TestBase
{
    [Fact]
    public void Run()
    {
        var assemblyPath = Path.Combine(CodeBaseLocation.CurrentDirectory, "DummyAssembly.dll");
        var weaver = new TargetWeaver();
        var result = weaver.ExecuteTestRun(assemblyPath);
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(result, ScrubCurrentDirectory);
#endif
    }

    static string ScrubCurrentDirectory(string s)
    {
        return s.Replace(@"\\", @"\").Replace(CodeBaseLocation.CurrentDirectory, "");
    }

    [Fact]
    public void WithCustomAssemblyName()
    {
        var assemblyPath = Path.Combine(CodeBaseLocation.CurrentDirectory, "DummyAssembly.dll");
        var weaver = new TargetWeaver();
        var result = weaver.ExecuteTestRun(
            assemblyPath: assemblyPath,
            assemblyName: "NewName");
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(result, ScrubCurrentDirectory);
#endif
    }
}

public class TargetWeaver : BaseModuleWeaver
{
    public override void Execute()
    {
        var type = FindType("System.Boolean");
        Assert.NotNull(type);
    }

    public override bool ShouldCleanReference => true;

    public override IEnumerable<string> GetAssembliesForScanning()
    {
        yield return "netstandard";
        yield return "mscorlib";
        yield return "System";
    }
}