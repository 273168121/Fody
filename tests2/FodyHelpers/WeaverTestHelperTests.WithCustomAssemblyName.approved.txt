{
  "Messages": [
    {
      "Text": "\tNo reference to 'Tests' found. References not modified.",
      "MessageImportance": "Normal"
    }
  ],
  "Warnings": [],
  "Errors": [],
  "Assembly": {
    "Data": "NewName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null",
    "UnityType": 6,
    "AssemblyName": "NewName, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null"
  },
  "AssemblyPath": "\fodytemp\NewName.dll"
}